Helper/Cart.php

```diff
@@ -711,7 +711,9 @@ public function getImmutableQuoteIdFromBoltOrder($response)
                 $immutableQuoteId = $response->cart->metadata->immutable_quote_id;
             } else {
                 // check if cart was created in plugin version before 2.14.0
-                list(, $immutableQuoteId) = explode(' / ', $response->cart->display_id);
+                if (isset($response->cart->display_id)) {
+                 list(, $immutableQuoteId) = explode(' / ', $response->cart->display_id);
+                }
```

Comment:
Is there anything we need to do here if it's _not_ set? The logic below this sends an exception for not having an immutable ID. Will this stay consistent in that things pre-2.14 will have a set display_id and things post-2.14 will have a set metadata?

Reply:
For post-2.14 we always have `$response->cart->metadata->immutable_quote_id` Fo pre-2.14 we always have `$response->cart->display_id`
So it's not a expected situation if we don't have both. I added this line just in case and to make tests more consistent.

Reply:
sounds good